torchvision/models/detection/_utils.py

```diff
@@ -270,6 +270,43 @@ def encode_single(self, reference_boxes: Tensor, proposals: Tensor) -> Tensor:
 
         return targets
 
+    def encode_all(self, reference_boxes: List[Tensor], proposals: List[Tensor]) -> Tensor:
+        """
+        vectorized version of `encode_single`
+        Args:
+            reference_boxes (Tensor): reference boxes
+            proposals (Tensor): boxes to be encoded
+
+        Returns:
+            Tensor: the encoded relative box offsets that can be used to
+            decode the boxes.
+
+        """
+
+        reference_boxes = torch.stack(reference_boxes)
```

Comment:
@abhi-glitchhg let me know if you agree with this approach? If so please do the required changes and from my side the PR is good to merge.

Reply:
Will update this pr in few hrs.

Reply:
your suggestion applies to the `decode_all` method also, but I am not touching it in this pr; there is a pr for cleanups of box_decoding, will do some changes in that.